plugins/inputs/exec/exec.go

```diff
@@ -49,20 +50,25 @@ type Exec struct {
 
 func NewExec() *Exec {
 	return &Exec{
-		runner:  CommandRunner{},
+		runner:  &CommandRunner{},
 		Timeout: config.Duration(time.Second * 5),
 	}
 }
 
 type Runner interface {
 	Run(string, []string, time.Duration) ([]byte, []byte, error)
+	SetDebug(bool)
 }
 
 type CommandRunner struct {
 	debug bool
 }
 
-func (c CommandRunner) truncate(buf bytes.Buffer) bytes.Buffer {
+func (c *CommandRunner) SetDebug(debug bool) {
+	c.debug = debug
+}
```

Comment:
Instead of using this function you could directly set the variable, can't you?

Reply:
Why not just initialize the runner in `Init()`?

Reply:
Sounds good to me, will update the pr